trainers/train_xgboost.py

```
@@ -0,0 +1,643 @@
+
+        X = df_smooth[self.feat_cols + smooth_cols].values.astype(float)
+        y = df_smooth["target"].values.astype(float)
+
+        train_mask, val_mask, test_mask = _time_masks(
+            df_smooth["date"],
+            self.config["training"]["val_start"],
+            self.config["training"]["test_start"],
+        )
+
+        X_train, y_train = X[train_mask], y[train_mask]
+        X_val, y_val = X[val_mask], y[val_mask]
```

Ruff (0.14.8)

Local variable `X_val` is assigned to but never used

Remove assignment to unused variable `X_val`

(F841)